### PR TITLE
fix: prometheus input plugin run outside k8s cluster err

### PR DESCRIPTION
**Pull request by yahaa**
### Required for all PRs:

- [ ] Updated associated README.md.
- [ ] Wrote appropriate unit tests.
- [ ] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves #

fix prometheus input plugin run outside k8s cluster err.

### CODE REVIEW

**Comment by sspaink** [comment]
@yahaa thanks for the pull request, can you add more detail on the error this is trying to resolve? 
I noticed you resolved the issue where watcher is being called before the error is checked, so that is great.

The CI is failing because `github.com/imdario/mergo` and `github.com/spf13/pflag` aren't in the right place in the `LICENSE_OF_DEPENDENCIES.md` it is kind of picky, it needs to be in alphabetical order.

**Comment by yahaa** [comment]
> @yahaa thanks for the pull request, can you add more detail on the error this is trying to resolve? I noticed you resolved the issue where watcher is being called before the error is checked, so that is great.
> 
> The CI is failing because `github.com/imdario/mergo` and `github.com/spf13/pflag` aren't in the right place in the `LICENSE_OF_DEPENDENCIES.md` it is kind of picky, it needs to be in alphabetical order.

[the first one](https://github.com/influxdata/telegraf/pull/9960/files#diff-dcebbfefbef8e986af6172ea7f96ab8b6d0b62aac383a7fa3f7bc1df5d06f2e0L57), when we run telegra in outside of a kubernetes cluster the line `config, err := rest.InClusterConfig()` will be always get an error.

[the second one](https://github.com/influxdata/telegraf/pull/9960/files#diff-dcebbfefbef8e986af6172ea7f96ab8b6d0b62aac383a7fa3f7bc1df5d06f2e0L50) using this method to get rest.Config instance will be cause kubernetes client [watch pod error](https://github.com/influxdata/telegraf/pull/9960/files#diff-dcebbfefbef8e986af6172ea7f96ab8b6d0b62aac383a7fa3f7bc1df5d06f2e0L109)

File: scripts/check-deps.sh [hunk 1]
```diff
@@ -55,9 +55,9 @@ done
 # github.com/foo/bar github.com/foo/bar/v3) there will be a duplicate
 # in the list.  Remove duplicates again.
 mv "${tmpdir}/HEAD" "${tmpdir}/HEAD-dup"
-uniq "${tmpdir}/HEAD-dup" > "${tmpdir}/HEAD"
+uniq "${tmpdir}/HEAD-dup" | sort -n > "${tmpdir}/HEAD"
 
-grep '^-' docs/LICENSE_OF_DEPENDENCIES.md | grep -v github.com/DataDog/datadog-agent | cut -f 2 -d' ' > "${tmpdir}/LICENSE_OF_DEPENDENCIES.md"
+grep '^-' docs/LICENSE_OF_DEPENDENCIES.md | grep -v github.com/DataDog/datadog-agent | cut -f 2 -d' ' | sort -n > "${tmpdir}/LICENSE_OF_DEPENDENCIES.md"
```
Comment (powersj) [inline scripts/check-deps.sh:60]: Why are these changes required for this PR? Seems odd to sort numerically.

**Review by powersj — CHANGES_REQUESTED** [review]
Hi,

Thanks for this PR! If you are still interested in landing this, can you please rebase this on master, resolve the conflicts, and answer my question about the check-deps changes?

Thanks!

File: scripts/check-deps.sh (hunk 1 above)
Comment (yahaa) [reply scripts/check-deps.sh]: Ok, Thanks for your feedback, i have rebase and resolve the conflicts.

 It's diffcult to add `github.com/imdario/mergo` and `github.com/spf13/pflag` in the right place in the `LICENSE_OF_DEPENDENCIES.md`  because we need to ensure it in alphabetical order.

Comment (powersj) [reply scripts/check-deps.sh]: Hmm `go mod tidy` did not handle that for you? I am still seeing a conflict.

Comment (yahaa) [reply scripts/check-deps.sh]: > @yahaa I'd prefer if we kept the dependency licenses in alphabetical order opposed to numerical, would you mind removing this change? We could always consider this in a separate pull request.

Sorry, forgot to reply you. I also think a single pr would be better.
